### ISSUE TITLE
Test dataset

### ISSUE THREAD

**Issue by twshimwell** [mhardcastle/ddf-pipeline#20]
Seems important to have a test dataset that we can fairly rapidly run through ddf-pipeline to test different parameters etc.

Thoughts on what we could get away with.
-- contains moderately bright source
-- contains diffuse structure
-- most flux within the inner 10,000x10,000 pixels for much faster imaging
-- averaged to 1ch/sb and 16s

**Comment by mhardcastle** [comment]
Agreed -- this would be good for DDF too. Happy for you to create such a thing and put it somewhere public -- selected from HETDEX dir-independent images perhaps?

**Comment by twshimwell** [comment]
Is being public essential for this? This nice galaxy cluster field arguably seems like the most ideal one  (nice bright source and faint diffuse structure) but its not in a region we will be making public

**Comment by mhardcastle** [comment]
Not essential, I was just thinking we could make it into a unit test for ddf-pipeline in which case it would be nice to be able to link to it from the github. Clearly at the moment nobody apart from us can run ddf-pipeline anyway so not a big deal in practice.

**Comment by twshimwell** [comment]
OK, well ill prepare these data and have a run through the pipeline with 10,000x10,000 pixels and see what kind of total runtime we have here.

**Comment by cyriltasse** [comment]
Discussing unit tests, since this is connected to kMS tests too, maybe we should sync with Oleg and his band since we discussed doing that a while back?

**Comment by mhardcastle** [comment]
Yes indeed. I didn't do anything about that at the time since we didn't have a suitable test dataset, but if Tim gets something sorted that might kill two birds with one stone.

(Going to be important if/when you merge our current DDF changes back into master.)

**Comment by twshimwell** [comment]
Yes would be nice to get an SKSP dataset used for unit tests. Making one completely public is fine.

**Comment by cyriltasse** [comment]
So we cout do two tests:

(i) DDF (SSD-DI)+kMS+DDF (SSD-DD_AP)
(ii) full ddf-pipeline run

then if (i) fails, no way (ii) will do it

**Comment by twshimwell** [comment]
sure could do that. If we changed (i) a little it could just be the first steps from ddf-pipeline (so DD_P rather than DD_AP). This ignores bootstrapping which is usually done after kMS but is fast.

**Comment by mhardcastle** [comment]
Yes. In fact maybe (i) is sufficient for the DDFacet test and (ii) is our internal ddf-pipeline unit test.

**Comment by twshimwell** [comment]
The image_dirin_SSD took about 1hr and killms takes 7mins per dataset. In comparison to the usual images (20,000x20,000 pixels and 2ch/sb 8s) the imaging is about 6-7 times faster and the calibration is 4 times faster. So I think we are looking at less than 24hrs for the full ddf-pipeline. Fast enough? Or can you think of other tweaks we can make to save time?

Here there are a few residuals coming into the image from brightish sources outside the new field of view and also the facets are of course much smaller than in the full run which will have an impact on the quality of the calibration.

**Comment by cyriltasse** [comment]
That's for a 10.000 pixels image? Impressive...

**Comment by twshimwell** [comment]
Indeed 10000x10000 very impressive! (note data 4 times smaller than usual too). 

Unless you know already then maybe in the future it is worth investigating how imaging time typically varies with the size? Perhaps our 20,000x20,000 default could be tweaked

**Comment by cyriltasse** [comment]
Should we think of cutting the field to 10000x10000 for the survey after a DI substraction? Would speed up the process enormously and we can have the new mosaic done by the end of the week? :)

**Comment by twshimwell** [comment]
We could revisit this again. We will need to subtract the sources outside that region (and think about at which step is best to do this) but that should just be some testing. I'll open another ticket :)

**Comment by twshimwell** [comment]
Another timing

Imaging the phase1 step whilst applying solutions (60 directions in this case) takes ~2.5hrs with 16s and 1ch/sb averaging and 10,000x10,000 pixels. By comparison the same step (also 60 directions) would have taken ~7hrs with 8s and 2ch/sb and 20,000x20,000 pixels. 

So some of the time difference evaporated but I guess because a decent amount of time is applying the solutions.

**Comment by twshimwell** [comment]
Hmmm seems like this is indeed very fast and well within 24hrs. However, with the standard ddf-pipeline and the dataset I am using the quality is not sufficiently good to precisely probe details of the calibration as there are too many sources outside the field of view.

**Comment by mhardcastle** [comment]
OK, so we see how Cyril gets on with subtraction.

**Comment by twshimwell** [comment]
I guess a test dataset is probably the next thing we should have so that we can continue to improve things. Given that the runtime now is only ~2.5-3.5 days I think we perhaps don't need to cut corners in the testing. I think perhaps one of the deep fields (probably Bootes) would make sense for a test dataset?

**Comment by mhardcastle** [comment]
Sure, sounds like a good idea to use Bootes. We will want this when the merge of Cyril's code with mainline DDF is done since there are features in that that it would be nice for us to use.

**Comment by twshimwell** [comment]
Which is your favourite Bootes dataset @cyriltasse

**Comment by twshimwell** [comment]
Perhaps given we might be moving towards big changes in the pipeline it might be nice to decide on this test dataset soonish.

I'd probably vote for A2256 or some other big cluster as thats the type of stuff we hope is better in Cyrils new pipeline.

**Comment by mhardcastle** [comment]
Thing is we want to test several things -- diffuse emission, bright sources, bad ionosphere... not sure one test dataset is going to do what we want -- which is perhaps why we haven't converged in the past year...
